inductor(CPU): add ISA check before do cpu fx packed weight

XiaobingSuper opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99601
* __->__ #99502

1. This PR is to fix https://github.com/pytorch/pytorch/issues/99423, which will add an ISA check before doing the bf16 weight pack.
2. Move CPU-related tests from ```test_torchinductor.py``` to ```test_cpu_repo.py``` to reduce the CI time.


cc @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

XiaobingSuper committed:
inductor(CPU): add ISA check before do cpu fx packed weight

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor(CPU): add ISA check before do cpu fx packed weight"


1. This PR is to fix https://github.com/pytorch/pytorch/issues/99423, which will add an ISA check before doing the bf16 weight pack.
2. Move CPU-related tests from ```test_torchinductor.py``` to ```test_cpu_repo.py``` to reduce the CI time.


cc jgong5 mingfeima sanchitintel ashokei jingxu10 soumith voznesenskym penguinwu anijain2305 EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jgong5 commented on aten/src/ATen/native/mkldnn/RegisterMkldnnOpContextClass.cpp:
```diff
@@ -6,12 +6,17 @@
 using namespace internal::convolution;
 
+bool is_mkldnn_bf16_supported() {
+  return mkldnn_bf16_device_check();
```
Handle arm case too? `mkldnn_bf16_device_check_arm`

XiaobingSuper replied:
yes, changed.

jgong5 commented on test/inductor/test_cpu_repro.py:
```diff
@@ -43,9 +43,12 @@
 
 
 class CPUReproTests(TestCase):
+    common = check_model
```
Other cases in this file can be simplified with `common` too.

XiaobingSuper replied:
let's do it at the next step.

XiaobingSuper committed:
Update on "inductor(CPU): add ISA check before do cpu fx packed weight"


1. This PR is to fix https://github.com/pytorch/pytorch/issues/99423, which will add an ISA check before doing the bf16 weight pack.
2. Move CPU-related tests from ```test_torchinductor.py``` to ```test_cpu_repo.py``` to reduce the CI time.


cc jgong5 mingfeima sanchitintel ashokei jingxu10 soumith voznesenskym penguinwu anijain2305 EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper committed:
Update on "inductor(CPU): add ISA check before do cpu fx packed weight"


1. This PR is to fix https://github.com/pytorch/pytorch/issues/99423, which will add an ISA check before doing the bf16 weight pack.
2. Move CPU-related tests from ```test_torchinductor.py``` to ```test_cpu_repo.py``` to reduce the CI time.


cc jgong5 mingfeima sanchitintel ashokei jingxu10 soumith voznesenskym penguinwu anijain2305 EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

XiaobingSuper commented:
@pytorchbot merge
